src/operations/search.rs

```diff
@@ -1,51 +1,195 @@
+    }
+}
+
+impl Printable for PackageSearchResult {
+    fn to_print_string(&self) -> String {
+        let repo = if &self.repo == "aur" {
+            (self.repo.clone() + "/").bold().cyan()
+        } else {
+            (self.repo.clone() + "/").bold().purple()
+        };
+        let name = &self.name.bold();
+        let version = &self.version.bold().green();
+        let groups = if self.groups.is_some() && !self.groups.as_ref().unwrap().is_empty() {
+            format!(" ({})", self.groups.clone().unwrap().join(", "))
+        } else {
+            "".to_string()
+        }
+        .bold()
+        .blue();
+        let out_of_date = if self.out_of_date.is_some() {
+            format!(
+                " [out of date: since {}]",
+                Local
+                    .timestamp(self.out_of_date.unwrap().try_into().unwrap(), 0)
+                    .date_naive()
+            )
+        } else {
+            "".to_string()
+        }
+        .bold()
+        .red();
+        let installed = if self.installed {
+            " [installed]".to_string()
+        } else {
+            "".to_string()
+        }
+        .bold()
+        .cyan();
+        let description = wrap_text(&self.description, 4).join("\n");
+
+        format!("{repo}{name} {version}{groups}{out_of_date}{installed}\n    {description}")
+    }
+}
 
 #[tracing::instrument(level = "trace")]
-pub async fn aur_search(query: &str, by_field: Option<SearchBy>, options: Options) {
+pub async fn aur_search(
+    query: &str,
+    by_field: Option<SearchBy>,
+    options: Options,
+) -> Vec<PackageSearchResult> {
+    let alpm = get_handler().unwrap();
+    let local = alpm.localdb();
     let packages = rpcsearch(query.to_string(), by_field.map(SearchBy::into))
         .await
         .silent_unwrap(AppExitCode::RpcError);
     let total_results = packages.len();
 
-    for package in &packages {
-        let description = if let Some(desc) = &package.description {
-            desc
-        } else {
-            ""
-        };
-        println!(
-            "aur/{} {}\n    {}",
-            package.name, package.version, description
-        )
-    }
-
     tracing::debug!("Found {total_results} resuls for \"{query}\" in AUR",);
+
+    let results: Vec<PackageSearchResult> = packages
+        .into_iter()
+        .map(|package| {
+            let name = package.name;
+            let version = package.version;
+            let groups = None;
+            let out_of_date = package.out_of_date;
+            let installed = local.pkg(&*name).is_ok();
+            let description = package.description;
+
+            PackageSearchResult {
+                repo: "aur".to_string(),
+                name,
+                version,
+                groups,
+                out_of_date,
+                installed,
+                description: description.unwrap_or_else(|| "No description".to_string()),
```

Comment:
Can we make the description optional and only put "No description" in the Display/Printable implementation?